checker/bin/infer-and-annotate.sh

```diff
@@ -0,0 +1,129 @@
+#!/bin/sh
+
+# This script runs the Checker Framework's signature inference
+# iteratively on a program, adding type annotations to the program, until the
+# .jaif files from one iteration are the same as the .jaif files from the
+# previous iteration (which means there is nothing new to infer anymore).
+
+# This script receives as arguments:
+# 1. Processor's name (in any form recognized by javac's -processor argument).
+# 2. Classpath (target project's classpath).
+# 3. Extra processor arguments which will be passed to the checker.
+# 4. List of paths to .java files in a program.
+
+# Example of usage:
+# ./infer-and-annotate.sh "LockChecker,NullnessChecker" \
+#     $JSR308/plume-lib/java/plume.jar -AprintErrorStack \
+#     `find $JSR308/plume-lib/java/src/plume/ -name "*.java"`
+
+# In case of using this script for Android projects, the classpath must include
+# paths to: android.jar, gen folder, all libs used, source code folder.
+# The Android project must be built with "ant debug" before running this script.
+
+# Halts the script when a nonzero value is returned from a command.
+set -e
+
+# Path to directory that will contain .jaif files after running the CF
+# with -AinferSignatures
+SIGNATURE_INFERENCE_DIR=build/signature-inference
+
+# Path that will contain .class files after running CF's javac. This dir will
+# deleted after executing this script.
+TEMP_DIR=build/temp-signature-inference-output
+
+# Path to directory will contain .jaif files from the previous interation.
+PREV_ITERATION_DIR=build/prev-signature-inference
+
+# Path to annotation-file-utilities.jar
+AFU_JAR="${CHECKERFRAMEWORK}/../annotation-tools/annotation-file-utilities/annotation-file-utilities.jar"
+
+# This function separates extra arguments passed to the checker from Java files
+# received as arguments.
+# TODO: Handle the following limitation: This function makes the assumption
+# that every argument starts with a hyphen. It means one cannot pass arguments
+# such as -processorpath and -source.
+read_input() {
+    processor=$1
+    cp=$2:$AFU_JAR
+    # Ignores first two arguments (processor and cp).
+    shift
+    shift
+    extra_args=""
+    java_files=""
+    for i in "$@"
+    do
+        # This function makes the assumption that every extra argument
+        # starts with a hyphen. The rest are .java files.
+        case "$1" in
+            -*)
+                extra_args="$extra_args $1"
+            ;;
+            *)
+                java_files="$java_files $1"
+            ;;
+        esac
+        shift
+    done
+}
+
+# Iteratively runs the Checker
+infer_and_annotate() {
+    DIFF_JAIF=firstdiff
+    # Creates signature-inference directory if it doesn't exist already.
+    # If it already exists, the existing .jaif files will be considered.
+    # This allows the user to provide some .jaif files as input.
```

Comment:
How can the user provide .jaif files as input?  Neither the manual nor the documentation of this script explains such functionality.